### PR TITLE
Rename `UiRect` to `Frame`

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

`UiRect` is confusingly named (nothing it represents is a rectangle) and we have a consensus (as far as I can tell) that `Frame` is the best alternative.

If `Frame` seems ambiguous (not so much internally but with concepts from foreign UI apis), `UiFrame` could also be okay.

Alternative proposal: #7710

## Changelog
* `UiRect` is renamed to `Frame`

## Migration Guide
`UiRect` has been renamed to `Frame`.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1077,7 +1077,7 @@ current changes on git with [previous release tags][git_tag_comparison].
 - [Expose mint feature in bevy_math/glam][5857]
 - [Utility methods for Val][6134]
 - [Register missing bevy_text types][6029]
-- [Add additional constructors for `UiRect` to specify values for specific fields][5988]
+- [Add additional constructors for `Frame` to specify values for specific fields][5988]
```
Comment (ItsDoot) [inline CHANGELOG.md:1080]: Likely didn't mean to edit the changelog.

Comment (ickshonpe) [reply CHANGELOG.md:1080]: Yep, used a lazy blanket search and replace to fix all the doc comments, that was silly 😅

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
@TimJentzsch @nicoburns for review please :)

File: crates/bevy_ui/src/geometry.rs [hunk 2]
```diff
@@ -132,22 +132,22 @@ pub struct UiRect {
...
 }
 
-impl UiRect {
+impl Frame {
     pub const DEFAULT: Self = Self {
         left: Val::DEFAULT,
         right: Val::DEFAULT,
         top: Val::DEFAULT,
         bottom: Val::DEFAULT,
     };
```
Comment (TimJentzsch) [inline crates/bevy_ui/src/geometry.rs:141]: I think it makes sense to either have no default at all or set this to `0`.
Not sure if that should be changed in this PR though.

**Review by TimJentzsch — APPROVED** [review]
Definitely better than `UiRect`:D

File: crates/bevy_ui/src/geometry.rs (hunk 2 above)
Comment (alice-i-cecile) [reply crates/bevy_ui/src/geometry.rs:141]: Yeah, I'd prefer to swap that later.

Comment (ickshonpe) [reply crates/bevy_ui/src/geometry.rs:141]: > I think it makes sense to either have no default at all or set this to `0`. Not sure if that should be changed in this PR though.

I think this is fixed in the remove `Val::Undefined` pr.

**Review by nicoburns — APPROVED** [review]
Looks good to me. And I like the name `Frame`.

**Comment by hymm** [comment]
Not sure about `Frame` seems like it'd be confused with the frame as in frames per second.

**Comment by alice-i-cecile** [comment]
Cart's previously stated that "Bevy is treated as a global namespace", which I think that `Frame` runs a bit close to violating. I still think it's a much better name than `UiRect`. `LayoutFrame` would also be fine by me, but I'm increasingly leaning towards distinct types for margin/padding/border.

**Comment by ickshonpe** [comment]
> Cart's previously stated that "Bevy is treated as a global namespace", which I think that `Frame` runs a bit close to violating. I still think it's a much better name than `UiRect`. `LayoutFrame` would also be fine by me, but I'm increasingly leaning towards distinct types for margin/padding/border.

Yeah in a global namespace `Frame` intuitively feels like it would be a type to do with animation, and it might be workable now to have only one type but as taffy adds more features the margin, border, and padding interfaces might diverge even further making distinct types even more desirable.

**Comment by ickshonpe** [comment]
I thought this would be an easily accepted non-controversial obvious improvement but I've come around to agreeing with hymm now. 

@alice-i-cecile 
Do you think I should close this, and we should just concentrate on #7710 (or an incremental step towards it, maybe just a separate `Margin` type to start with)?

**Comment by ickshonpe** [comment]
Thinking on it, maybe #7710 and #7569 don't make sense as distinct PRs and should be combined.

**Comment by alice-i-cecile** [comment]
I think close this out and merge the other two :)

**Comment by viridia** [comment]
BTW, the word you are looking for is `Insets`. That's what it's called in iOS (`UiEdgeInsets`), Android (`Insets`) and Flutter (`EdgeInsets`).

@alice-i-cecile @ickshonpe

**Pull request closed without merging**